### ISSUE TITLE
[Feature request] tf.edit_distance return deletions, substitutions, insertions

### ISSUE THREAD

**Issue by georgesterpu** [tensorflow/tensorflow#14252]
Hi,
When analysing the performance of a speech recognition model, it is very useful to know the distribution of deletions, substitutions and insertions on the test set.

Could you make `tf.edit_distance` return these three metrics too, in addition to the cheapest cost?
It appears to me that the [cpp code](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/lib/gtl/edit_distance.h#L47) already computes them. What do you think, @ebrevdo ?

**Comment by ebrevdo** [comment]
Sounds like we need to create a new op that returns all three

**Comment by georgesterpu** [comment]
Hi @ebrevdo ,
Excuse me for this rookie question.
How do I find the gradient implementation of tf.edit_distance (or any other op) ?

I added the tensor returned by tf.edit_distance to the traditional cross entropy loss, and can see that it is possible to pass this to an optimiser. The aim is to obtain an implementation that is conformant with Google's paper https://arxiv.org/abs/1712.01818

Thank you

**Comment by ebrevdo** [comment]
edit_distance does not have a gradient, because it has integer outputs and
outputs and is not differentiable.  only ctc_loss is differentiable.

**Comment by colmantse** [comment]
hi @ebrevdo , since this issue has been closed, just wondering if the feature has been implemented. thanks much!

**Comment by andreselizondo-adestech** [comment]
@ebrevdo Would it be possible to get a gradient for edit_distance when the output is normalized?
This way the output would be a float.
